optuna/integration/pytorch_lightning.py

```diff
@@ -60,7 +61,9 @@ def on_init_start(self, trainer: Trainer) -> None:
         if self.is_ddp_backend:
             if version.parse(pl.__version__) < version.parse("1.5.0"):
                 raise ValueError("PyTorch Lightning>=1.5.0 is required in DDP.")
-            if not isinstance(self._trial.study._storage, _CachedStorage):
+            if (not isinstance(self._trial.study._storage, _CachedStorage)) or (
+                not isinstance(self._trial.study._storage._backend, RDBStorage)
+            ):
```

Comment:
```suggestion
            if not (
                isinstance(self._trial.study._storage, _CachedStorage)
                and isinstance(self._trial.study._storage._backend, RDBStorage)
            ):
```

How about using transformed version?

Reply:
Thanks, it is more clear. Fixed.